src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -763,6 +763,19 @@ object TezosDatabaseOperations extends LazyLogging {
   def writeProcessedEventsLevels(eventType: String, levels: List[BigDecimal]): DBIO[Option[Int]] =
     Tables.ProcessedChainEvents ++= levels.map(Tables.ProcessedChainEventsRow(_, eventType))
 
+  /** Adds any new level for which a custom event processing has been executed
+    *
+    * @param rows the type of event to record
+    * @return the number of entries saved added to the
+    */
+  def writeRegisteredTokensRowsIfEmpty(
+      rows: List[Tables.RegisteredTokensRow]
+  )(implicit ec: ExecutionContext): DBIO[Option[Int]] =
+    Tables.RegisteredTokens.length.result.flatMap {
```

Comment:
There's a `RegisteredTokens.exists` method that will tell if there's any row, and I suspect it to be much more efficient than counting all the rows.

Reply:
ok, used exists, good catch